models/project_repository_blackbox_test.go

```diff
@@ -0,0 +1,173 @@
+package models_test
+
+import (
+	"testing"
+
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/almighty/almighty-core/models"
+	"github.com/almighty/almighty-core/project"
+	satoriuuid "github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"golang.org/x/net/context"
+)
+
+var testProject string = satoriuuid.NewV4().String()
+var testProject2 string = satoriuuid.NewV4().String()
+
+func TestRunProjectRepoBBTest(t *testing.T) {
+	suite.Run(t, &projectRepoBBTest{DBTestSuite: gormsupport.NewDBTestSuite("../config.yaml")})
+}
+
+type projectRepoBBTest struct {
+	gormsupport.DBTestSuite
+	undoScript *gormsupport.DBScript
+	repo       *models.UndoableProjectRepository
+}
+
+func (test *projectRepoBBTest) SetupTest() {
+	test.undoScript = &gormsupport.DBScript{}
+	test.repo = models.NewUndoableProjectRepository(models.NewProjectRepository(test.DB), test.undoScript)
+	test.DB.Unscoped().Delete(&project.Project{}, "Name=?", testProject)
+	test.DB.Unscoped().Delete(&project.Project{}, "Name=?", testProject2)
+}
+
+func (test *projectRepoBBTest) TearDownTest() {
+	test.undoScript.Run(test.DB)
+}
+
+func (test *projectRepoBBTest) TestCreate() {
```

Comment:
I think I generally would prefer Test names that try to explain What the test scenario is, more than the Code that is being tested.

e.g.

* TestCreateFailOnEmptyName
* TestCreateValidProject
* TestCreateFailOnDuplicateProject